pkg/cluster/rook_ceph.go

```diff
@@ -824,10 +824,31 @@ func (c *Controller) cephOSDPoolSetSize(name string, size int, cephcluster *ceph
 			}
 		}
 	}
-	return c.rookCephExec(args...)
+	return c.rookCephExec(rookVersion, args...)
 }
 
-func (c *Controller) cephOSDPoolSetMinSize(name string, minSize int) error {
+func (c *Controller) cephOSDPoolSetMinSize(rookVersion semver.Version, name string, minSize int) error {
 	args := []string{"ceph", "osd", "pool", "set", name, "min_size", strconv.Itoa(minSize)}
-	return c.rookCephExec(args...)
+	return c.rookCephExec(rookVersion, args...)
+}
+
+// GetRookVersion gets the Rook version from the container image tag of the rook-ceph-operator
+// deployment in the rook-ceph namespace.
+func (c *Controller) GetRookVersion(ctx context.Context) (*semver.Version, error) {
+	deploy, err := c.Config.Client.AppsV1().Deployments("rook-ceph").Get(ctx, "rook-ceph-operator", metav1.GetOptions{})
+	if err != nil {
+		return nil, errors.Wrap(err, "get rook-ceph-operator deployment")
+	}
+	for _, container := range deploy.Spec.Template.Spec.Containers {
+		if container.Name == "rook-ceph-operator" {
+			parts := strings.Split(deploy.Spec.Template.Spec.Containers[0].Image, ":")
+			imageTag := parts[len(parts)-1]
+			rookVersion, err := semver.Parse(strings.TrimPrefix(imageTag, "v"))
+			if err != nil {
+				return nil, errors.Wrap(err, "parse semver")
+			}
+			return &rookVersion, nil
+		}
+	}
```

Comment:
Another option is to inspect the deployment's label and get the version from helm chart label: `helm.sh/chart: rook-ceph-v1.9.12`, it would simplify the code a bit. However, getting the rook version from the label is not as durable since someone/something can remove the label and break this logic.

Reply:
also its possible we change how we generate the add-on in the future and no longer use helm